tool/tsh/common/tsh.go

```diff
@@ -2923,81 +2936,63 @@ func getDatabaseRow(proxy, cluster, clusterFlag string, database types.Database,
 		}
 	}
 
-	row := make([]string, 0)
-	if proxy != "" && cluster != "" {
-		row = append(row, proxy, cluster)
-	}
-
-	labels := common.FormatLabels(database.GetAllLabels(), verbose)
-	if verbose {
-		row = append(row,
-			displayName,
-			database.GetDescription(),
-			database.GetProtocol(),
-			database.GetType(),
-			database.GetURI(),
-			formatUsersForDB(database, accessChecker),
-			labels,
-			connect,
-		)
-	} else {
-		row = append(row,
-			displayName,
-			database.GetDescription(),
-			formatUsersForDB(database, accessChecker),
-			labels,
-			connect,
-		)
+	// Must match printDatabaseTableConfig.allColumnTitles().
```

Comment:
Yikes. I know that this is the approach elsewhere, but I wonder if we can have a struct to hold these values. For example:

```golang
type databaseDescriptionRow struct {
	Proxy         string
	Cluster       string
	Name          string
	Description   string
	Protocol      string
	Type          string
	URI           string
	AllowedUsers  string
	DatabaseRoles string
	Labels        string
	Connect       string
}

func dbDescriptionColumns() []string {
	var out []string
	// regular expression to convert from "DatabaseRoles" to "Database Roles" etc.
	re := regexp.MustCompile(`([a-z])([A-Z])`)
	t := reflect.TypeOf(databaseDescriptionRow{})
	for i := 0; i < t.NumField(); i++ {
		out = append(out, re.ReplaceAllString(t.Field(i).Name, "${1} ${2}"))
	}
	return out
}

func (row databaseDescriptionRow) Columns() []string {
	var out []string
	val := reflect.ValueOf(row)
	for i := 0; i < val.NumField(); i++ {
		out = append(out, val.Field(i).String())
	}
	return out
}
```

Reply:
I started with exactly this struct but I didn't want to do too much refactoring. But since asked, let me give it another try

Reply:
Thanks! I know this is a balancing act, so I appreciate the effort; if it doesn't work out, so be it.